src/client/components/pages/Courses/MeetingTimesList.tsx

```diff
@@ -412,46 +334,23 @@ export const MeetingTimesList
           )
         )}
       </ul>
-      <VerticalSpace />
-      <ButtonLayout>
+      <StyledButtonLayout>
         <Button
           id="addNewTimeButton"
-          onClick={
-            (): void => {
-              // Close the current meeting only if there are no validation errors
-              if (validateTimes()) {
-                // If there is a meeting currently being edited, get the data
-                // currently being edited and call onChange to update the meetings
-                let updatedMeetings = meetings;
-                if (currentMeetingId != null) {
-                  updatedMeetings = getCurrentMeetings();
-                  setCurrentMeetingId(null);
-                }
-                // Generate a new meeting row using a newly created meeting ID via the newMeetingIndex
-                const newMeeting = {
-                  id: `newMeeting${newMeetingIndex}`,
-                  day: '' as DAY,
-                  startTime: '',
-                  endTime: '',
-                  room: null,
-                };
-                updatedMeetings = [...updatedMeetings, newMeeting];
-                // update the meetings
-                onChange(updatedMeetings);
-                setCurrentMeetingId(newMeeting.id);
-                setNewMeetingIndex(newMeetingIndex + 1);
-                setCurrentDay(newMeeting.day);
-                setCurrentStartTime(newMeeting.startTime);
-                setCurrentEndTime(newMeeting.endTime);
-                setCurrentRoom(newMeeting.room);
-              }
-            }
-          }
+          onClick={() => {
+            closeCurrentEditMeeting({
+              id: `new-meeting-${allMeetings.length + 1}`,
+              day: '' as DAY,
+              startTime: '',
+              endTime: '',
+              room: null,
```

Comment:
Is there a reason I'm not immediately seeing for this object being passed in here like this instead of `null` like elsewhere? All the other instances are either passed an argument of type `CourseInstanceResponseMeeting` or `null`.

Reply:
This is just passing in a blank `CourseInstanceResponseMeeting` object, which handles the case of adding a new meeting. The other two cases you mentioned, `null` and existing meeting, are for the cases of closing all the meetings and editing an existing meeting, respectively.

Reply:
Ah, I see. Yeah, it just suddenly clicked. Got it.